### ISSUE TITLE
Events fail to initialize on `react@^18`

### ISSUE THREAD

**Issue by nickofthyme** [elastic/elastic-charts#2171]
When using charts with `react@^18`, the event callbacks for user interactions all fail to be called.

> With React18 that seriously blocks us: event callbacks like `onBrushEnd` are not called in React18.
> 
> _Originally posted by @yannbolliger in https://github.com/elastic/elastic-charts/issues/1910#issuecomment-1715683028_

This issue appears to be related to the `ResizeObserver` calling `onResize` with an empty element (i.e. `{ height: 0, width: 0 }`). Then the `UPDATE_PARENT_DIMENSION` is updating to the zeroed dimensions and blocking the initialization of the event callbacks.

**Comment by nickofthyme** [comment]
@markov00 I took a look into this and for some reason the redux state does not appear to be updating the parent dimensions. The `UPDATE_PARENT_DIMENSION` reducer is called with a non-zero `height` and `width` but then immediately after the `getInternalIsInitializedSelector` returns a non-initialized state.

When you get a chance could you take a look, cuz I'm out of ideas on what it could be.

**Comment by markov00** [comment]
> @markov00 I took a look into this and for some reason the redux state does not appear to be updating the parent dimensions. The `UPDATE_PARENT_DIMENSION` reducer is called with a non-zero `height` and `width` but then immediately after the `getInternalIsInitializedSelector` returns a non-initialized state with `height = width = 0`.
> 
> When you get a chance could you take a look, cuz I'm out of ideas on what it could be.

I did but not fully sure about my findings (I need to dig a bit more):
- it seems related to the way we dispatch the call: we update from withing a RAF
- it could be also related to the debounce function used: react 18 put[ some changes to how setTimeout works](https://react.dev/blog/2022/03/29/react-v18#new-feature-automatic-batching), not sure if this is related but we should check

**Comment by nickofthyme** [comment]
This issue appears to be caused by using the [`StrictMode`](https://react.dev/reference/react/StrictMode) component from `react`. Still investigating why this is...

**Comment by nickofthyme** [comment]
Since kibana does not use `StrictMode` I am pushing this fix to `8.16` cycle to be done along with work to upgrade redux https://github.com/elastic/elastic-charts/issues/2377.

**Comment by jonas080301** [comment]
Hi there, i am using elastic/charts with Next13 and have the same error with onElementClick event. Is there any workaround or fix?

**Comment by nickofthyme** [comment]
@jonas080301 are you using `StrictMode`?

**Comment by jonas080301** [comment]
> @jonas080301 are you using `StrictMode`?

Looks like NextJS uses React in Strict Mode. I got it to work by setting reactStrictMode to false in the next.config.ts! Thank you

**Comment by nickofthyme** [comment]
Glad to hear! I plan to dig deeper into the issue soon but for now that is the best work around.

**Comment by markov00** [comment]
@nickofthyme We should remember that this still now working even now that we have Moved to react 18.

**Comment by nickofthyme** [comment]
Yeah even after the redux migration 😔. We'll have to investigate it again at some point.
